### PR TITLE
Improve Code coverage and add tests

### PR DESCRIPTION
**Pull request by Maasdamind**
# How was this tested?

* Step one etc.

# Type of change

Please check all boxes that apply to this change below.

| Type                     | Applies ✅|
| ------------------------ | :-------: |
| Backend                  |           |
| Frontend                 |           |
| Github/Ci/Admin          |           |
| Documentation required   |           |

# Checklist

Please follow the checklist below and check all completed items as you work on the PR.

I have..
- [x] performed a self-review of my own code
- [ ] commented my code, particularly in hard-to-understand areas
- [ ] written documentation
- [x] added automated tests
- [ ] merged dependencies to `develop

### CODE REVIEW

File: .github/workflows/code-coverage.yml [hunk 1]
```diff
@@ -19,11 +19,11 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v2
         with:
-          fetch-depth: 2
+          fetch-depth: 0
```
Comment (pizzaluc) [inline .github/workflows/code-coverage.yml:22]: Why is this changed to 0?

Comment (Maasdamind) [reply .github/workflows/code-coverage.yml:22]: So it fetches all commits. A bit more costly, but not really since we don't have a lot of traffic

Comment (pizzaluc) [reply .github/workflows/code-coverage.yml:22]: Oh i see, i thought higher depth meant more thourough fetching

Comment (Maasdamind) [reply .github/workflows/code-coverage.yml:22]: i believe it does, but i believe 0 means all :lab_coat:

**Review by tjomson — APPROVED** [review]
If Lucas has no more comments, i think this looks good as well 😃
